packages/astro/src/runtime/server/astro-global.ts

```diff
@@ -25,29 +18,15 @@ function createAstroGlobFn() {
 
 // This is used to create the top-level Astro global; the one that you can use
 // Inside of getStaticPaths.
+// TODO: remove `_filePathname` and `_projectRootStr` from the compiler
 export function createAstro(
-	filePathname: string,
-	_site: string | undefined,
-	projectRootStr: string
+	_filePathname: string,
+	site: string | undefined,
+	_projectRootStr: string
 ): AstroGlobalPartial {
-	const site = _site ? new URL(_site) : undefined;
-	const referenceURL = new URL(filePathname, `http://localhost`);
-	const projectRoot = new URL(projectRootStr);
 	return {
-		site,
+		site: site ? new URL(site) : undefined,
 		generator: `Astro v${ASTRO_VERSION}`,
-		fetchContent: createDeprecatedFetchContentFn(),
 		glob: createAstroGlobFn(),
-		// INVESTIGATE is there a use-case for multi args?
-		// TODO remove in 2.0
-		resolve(...segments: string[]) {
```

Comment:
Ah `Astro.resolve` I mind when you came in, and then boom it was used everywhere, prior to the shift to esm. 
RIP